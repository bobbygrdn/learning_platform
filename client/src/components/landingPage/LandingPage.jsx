/* eslint-disable react/jsx-no-comment-textnodes */
import React, { useEffect } from 'react'
import Header from './Header'
import CallToAction from './CallToAction';
import Button from './Button'
import UserReviews from './UserReviews'
import KeyFeatures from './KeyFeatures'
import Overview from './Overview'
import GettingStarted from './GettingStarted'
import Footer from './Footer'
import Demo from './Demo';
import WarriorCards from './WarriorCards';
import '../../styles/landingPage/LandingPage.css'

export default function LandingPage() {
    const [reviews, setReviews] = React.useState([])

    useEffect(() => {
        const fetchData = async () => {
            const response = await fetch('/dummyData/reviews.json');
            const data = await response.json();
            setReviews(data);
        };
        fetchData();
    }, [])

    const handleClick = () => {

    }

    return (
        <>
            <Header />
            <WarriorCards />
            <CallToAction />
            <Button purpose={"landingPage"} onClick={handleClick} text={"Start Your Journey"} />
            <UserReviews reviews={reviews} />
            <KeyFeatures />
            <Overview />
            <GettingStarted />
            {/* TODO: Add demo video url*/}
            <Demo video={""} />
            <Footer />
        </>
    )
}
