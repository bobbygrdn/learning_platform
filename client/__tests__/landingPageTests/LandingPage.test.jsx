import React from 'react';
import { render, screen } from '@testing-library/react';
import LandingPage from '../../src/components/landingPage/LandingPage';

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useEffect: () => { },
}))

it('renders without crashing', () => {
    const { container } = render(<LandingPage />);
    expect(container).toMatchSnapshot();
})

it('should render a header component', () => {
    render(<LandingPage />);
    expect(screen.getAllByTestId('header')).toHaveLength(1);
})

it('should render a list of warrior names', () => {
    render(<LandingPage />);
    expect(screen.getAllByTestId('warriorList')).toHaveLength(1);
})

it('should render a warrior name', () => {
    render(<LandingPage />);
    expect(screen.getAllByTestId('warriorName')).toHaveLength(4);
})

it('should render warrior cards', () => {
    render(<LandingPage />);
    expect(screen.getAllByTestId('card')).toHaveLength(4);
})

it('should render call to action statements', () => {
    render(<LandingPage />);
    expect(screen.getAllByTestId('callToAction')).toHaveLength(1);
})

it('should render button component', () => {
    render(<LandingPage />);
    expect(screen.getAllByTestId('landingPage')).toHaveLength(1);
})

it('should render user reviews component', () => {
    render(<LandingPage />);
    expect(screen.getAllByTestId('reviews')).toHaveLength(1);
})

it('should render key features component', () => {
    render(<LandingPage />);
    expect(screen.getAllByTestId('features')).toHaveLength(1);
})

it('should render overview component', () => {
    render(<LandingPage />);
    expect(screen.getAllByTestId('overview')).toHaveLength(1);
})

it('should render getting started component', () => {
    render(<LandingPage />);
    expect(screen.getAllByTestId('steps')).toHaveLength(1);
})

it('should render demo component', () => {
    render(<LandingPage />);
    expect(screen.getAllByTestId('demo-video')).toHaveLength(1);
})

it('should render a footer component', () => {
    render(<LandingPage />);
    expect(screen.getAllByTestId('footer')).toHaveLength(1);
})