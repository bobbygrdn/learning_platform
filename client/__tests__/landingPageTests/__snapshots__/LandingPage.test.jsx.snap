// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <header
    class="header"
    data-testid="header"
  >
    <div
      class="titles"
    >
      <p
        class="title"
        data-testid="title"
      >
        Valor Academy
      </p>
      <p
        class="sectiontitle"
        data-testid="sectiontitle"
      >
        Become a Code Warrior
      </p>
    </div>
    <img
      alt="Valor Academy logo"
      class="logo"
      data-testid="logo"
      src="undefined/resources/valorAcademy_logo.png"
    />
  </header>
  <div
    class="warriorCards"
  >
    <ul
      class="warriorList"
      data-testid="warriorList"
    >
      <li
        class="warriorName"
        data-testid="warriorName"
      >
        Samurai
      </li>
      <li
        class="warriorName"
        data-testid="warriorName"
      >
        Legionnaire
      </li>
      <li
        class="warriorName"
        data-testid="warriorName"
      >
        Spartan
      </li>
      <li
        class="warriorName"
        data-testid="warriorName"
      >
        Viking
      </li>
    </ul>
    <div
      class="warriors"
    >
      <img
        alt="warrior image"
        class="card"
        data-testid="card"
        src=""
      />
      <img
        alt="warrior image"
        class="card"
        data-testid="card"
        src=""
      />
      <img
        alt="warrior image"
        class="card"
        data-testid="card"
        src=""
      />
      <img
        alt="warrior image"
        class="card"
        data-testid="card"
        src=""
      />
    </div>
  </div>
  <div
    class="callToAction"
    data-testid="callToAction"
  >
    <p
      class="subheading"
      data-testid="subheading"
    >
      Choose your path to eternal glory and fame
    </p>
    <p
      class="subheading"
      data-testid="subheading"
    >
      Take on challenges, level up your knowledge and become a code warrior legend
    </p>
    <p
      class="subheading"
      data-testid="subheading"
    >
      Discover a better way to become triumphant in your coding journey
    </p>
  </div>
  <button
    class="landingPage"
    data-testid="landingPage"
  >
    Start Your Journey
  </button>
  <div
    class="reviews"
    data-testid="reviews"
  />
  <div
    class="features"
    data-testid="features"
  >
    <div
      class="feature"
      data-testid="feature"
    >
      <img
        alt="Feature undefined"
        class="icon"
        src="undefined/resources/learningPath_icon.png"
      />
      <p
        class="text"
        data-testid="text"
      >
        Custom learning paths for different goals and skill levels
      </p>
    </div>
    <div
      class="feature"
      data-testid="feature"
    >
      <img
        alt="Feature undefined"
        class="icon"
        src="undefined/resources/library_icon.png"
      />
      <p
        class="text"
        data-testid="text"
      >
        Access a library of courses created by expert programmers
      </p>
    </div>
    <div
      class="feature"
      data-testid="feature"
    >
      <img
        alt="Feature undefined"
        class="icon"
        src="undefined/resources/forum_icon.png"
      />
      <p
        class="text"
        data-testid="text"
      >
        Engage with a community of fellow learners and experts
      </p>
    </div>
    <div
      class="feature"
      data-testid="feature"
    >
      <img
        alt="Feature undefined"
        class="icon"
        src="undefined/resources/mobile_icon.png"
      />
      <p
        class="text"
        data-testid="text"
      >
        Mobile-first design for on-the-go learning
      </p>
    </div>
    <div
      class="feature"
      data-testid="feature"
    >
      <img
        alt="Feature undefined"
        class="icon"
        src="undefined/resources/certificate_icon.png"
      />
      <p
        class="text"
        data-testid="text"
      >
        Earn badges and certificates to demonstrate your knowledge and progress
      </p>
    </div>
    <div
      class="feature"
      data-testid="feature"
    >
      <img
        alt="Feature undefined"
        class="icon"
        src="undefined/resources/recommendations_icon.png"
      />
      <p
        class="text"
        data-testid="text"
      >
        Personalized recommendations based on your learning history and preferences
      </p>
    </div>
  </div>
  <div
    class="overview"
    data-testid="overview"
  >
    <p
      class="callStatement"
      data-testid="callStatement"
    >
      Welcome to Valor Academy!
    </p>
    <p
      class="overviewText"
      data-testid="overviewText"
    >
      This platform was created to engage and help guide the developer community into a new style of learning. Our focus is to allow developers in all stages of their journey to grow and become more confident in their abilities. As you progress through the different courses, lessons, quizzes and more, you will see that we focus on providing great content and a roadmap for you to level up!
    </p>
  </div>
  <p
    class="steps"
    data-testid="steps"
  >
    Here is a step by step guide to get you started!
  </p>
  <div
    class="gettingStarted"
  >
    <img
      alt="Step 1"
      class="step"
      data-testid="step"
      src="/resources/male_meijin.jpg"
    />
    <img
      alt="Arrow"
      class="arrow arrow1"
      data-testid="arrow"
      src="/resources/arrow_right.png"
    />
    <img
      alt="Step 2"
      class="step"
      data-testid="step"
      src="/resources/male_meijin.jpg"
    />
    <img
      alt="Arrow"
      class="arrow arrow2"
      data-testid="arrow"
      src="/resources/curved_arrow_right.png"
    />
    <img
      alt="Step 3"
      class="step"
      data-testid="step"
      src="/resources/male_meijin.jpg"
    />
    <img
      alt="Arrow"
      class="arrow arrow3"
      data-testid="arrow"
      src="/resources/curved_arrow_left.png"
    />
    <img
      alt="Step 4"
      class="step"
      data-testid="step"
      src="/resources/male_meijin.jpg"
    />
    <img
      alt="Arrow"
      class="arrow arrow4"
      data-testid="arrow"
      src="/resources/arrow_right.png"
    />
    <img
      alt="Step 5"
      class="step"
      data-testid="step"
      src="/resources/male_meijin.jpg"
    />
  </div>
  <div
    data-testid="demo-video"
    style="width: 640px; height: 360px;"
  />
  <footer
    data-testid="footer"
  >
    <p
      class="copyright"
      data-testid="copyright"
    >
      Copyright Â© 2023 Valor Academy. All rights reserved.
    </p>
    <p
      class="media"
      data-testid="media"
    >
      Follow us on social media for news, updates, and promotions.
    </p>
    <ul
      data-testid="ul"
    >
      <li
        data-testid="li"
      >
        <a
          data-testid="anchor"
          href="/terms"
        >
          Terms of Service
        </a>
      </li>
      <li
        data-testid="li"
      >
        <a
          data-testid="anchor"
          href="/privacy"
        >
          Privacy Policy
        </a>
      </li>
      <li
        data-testid="li"
      >
        <a
          data-testid="anchor"
          href="https://github.com/bobbygrdn/robert_gordon_capstone.git"
        >
          GitHub
        </a>
      </li>
      <li
        data-testid="li"
      >
        <a
          data-testid="anchor"
          href="https://bobbygrdn.atlassian.net/wiki/spaces/VA/pages/7012354/Documentation?atlOrigin=eyJpIjoiODRkZmIzNWU2YjIxNGEyYjliZDAyMTVlYjExN2QyN2QiLCJwIjoiaiJ9"
        >
          Jira
        </a>
      </li>
    </ul>
  </footer>
</div>
`;
